pytorch_lightning/core/lightning.py

```diff
@@ -327,6 +327,67 @@ def log_dict(
                 tbptt_reduce_fx=tbptt_reduce_fx,
             )
 
+    def add_predictions(self, predictions: Union[torch.Tensor, List]) -> None:
+        """
+        This function enables to save add predictions and retrieved them into results objects.
+
+        .. code-block:: python
+
+            import pandas as pd
+
+            class MyModel(LightningModule):
+
+                # VERSION 1
+                def test_step(self, batch, batch_idx, dataloader_idx=None):
+                    x, y = batch
+                    output = self.layer(x)
+                    loss = self.loss(y, output)
+
+                    # Add directly a tensor
+                    self.add_predictions(output)
+
+                    return {"y": loss}
+
+                # VERSION 2
+                def test_step(self, batch, batch_idx, dataloader_idx=None):
+                    x, y = batch
+                    output = self.layer(x)
+                    loss = self.loss(y, output)
+
+                    # Support list of dictionaries.
+                    self.add_predictions([{"pred": o, "target": t} for o, t in zip(output, y)])
+
+                    return {"y": loss}
+
+            results = trainer.test(test_dataloaders=test_dataloaders)
```

Comment:
I thought the interface would be Trainer.predict using forward only not the test approach using test_step?

Reply:
I would work on `.predict` in another PR. But if the user provides raw_data, then predict is really a pipeline and there is much more work to make this possible.